api/v2/pinning.go

```diff
@@ -0,0 +1,172 @@
+	"github.com/application-research/estuary/util"
+	"github.com/ipfs/go-cid"
+	"github.com/labstack/echo/v4"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"golang.org/x/xerrors"
+	"gorm.io/gorm"
+	"net/http"
+	"path/filepath"
+)
+
+type BatchedPinRequest struct {
+	ContentIdToPin string `json:"content_id"`
+}
+
+// handleGetBatchedPins  godoc
+// @Summary      Get a pin status object
+// @Description  This endpoint returns a pin status object.
+// @Tags         pinning
+// @Produce      json
+// @Success      200	{object}  []types.IpfsPinStatusResponse
+// @Failure      404	{object}  util.HttpError
+// @Failure      500    {object}  util.HttpError
+// @Param        pin           body      []api.BatchedPinRequest  true   "Pin Body {[content_id:"content_id_to_pin"]}"
+// @Router       /v2/pinning/batched-pins/ [get]
+func (s *apiV2) handleGetBatchedPins(c echo.Context, u *util.User) error {
+
+	var pins []BatchedPinRequest
+	err := c.Bind(&pins)
+	if err != nil {
+		return err
+	}
+
+	var pinStatuses []*types.IpfsPinStatusResponse
+	for _, pinId := range pins {
+		contentId := pinId.ContentIdToPin
+		var content util.Content
+		if err := s.DB.First(&content, "id = ? AND not replace", contentId).Error; err != nil {
+			if xerrors.Is(err, gorm.ErrRecordNotFound) {
+				return &util.HttpError{
+					Code:    http.StatusNotFound,
+					Reason:  util.ERR_CONTENT_NOT_FOUND,
+					Details: fmt.Sprintf("Content with id %s not found", contentId),
+				}
+			}
+			return err
+		}
+
+		if err := util.IsContentOwner(u.ID, content.UserID); err != nil {
+			return err
+		}
+
+		st, err := s.CM.PinStatus(content, nil)
+		pinStatuses = append(pinStatuses, st)
+		if err != nil {
+			return err
+		}
+	}
+
+	return c.JSON(http.StatusOK, pinStatuses)
+}
+
+// handleAddBatchedPins  godoc
+// @Summary      Add and pin objects in batches
+// @Description  This endpoint adds a pin to the IPFS daemon.
+// @Tags         pinning
+// @Accept		 json
+// @Produce      json
+// @Success      202	{object}  []types.IpfsPinStatusResponse
+// @Failure      500    {object}  util.HttpError
+// @in           202,default  string  Token "token"
+// @Param        pin           body      []types.IpfsPin  true   "Pin Body {[cid:cid, name:name]}"
+// @Param        ignore-dupes  query     string                   false  "Ignore Dupes"
+// @Param        overwrite	   query     string                   false  "Overwrite conflicting files in collections"
+// @Router       /v2/pinning/batched-pins [post]
+func (s *apiV2) handleAddBatchedPins(c echo.Context, u *util.User) error {
+	ctx := c.Request().Context()
+
+	if err := util.ErrorIfContentAddingDisabled(s.isContentAddingDisabled(u)); err != nil {
+		return err
+	}
+
+	overwrite := false
+	if c.QueryParam("overwrite") == "true" {
+		overwrite = true
+	}
+
+	var pins []types.IpfsPin
+	if err := c.Bind(&pins); err != nil {
+		return err
+	}
+
+	var pinStatuses []*types.IpfsPinStatusResponse
+	for _, pin := range pins {
+		filename := pin.Name
```

Comment:
@alvin-reyes can we abstract these lines below and reuse for both single and batch